resources/common/plugins/template/cvvc-oto-gen/cvvc-oto-gen.js

```diff
@@ -117,136 +112,322 @@ let aliasCountMap = new Map()
+    
+    let end = start - cutoff
+    let points = [fixed, preu, ovl]
+    points.push(start)
+    let extras = [cutoff.toString()]
+    let notes = new Notes()
+    if (appendTags) {
+        notes.tag = "vcv"
+    }
+    let entry = new Entry(sample, thisAlias, start, end, points, extras, notes)
+    push(entry, "other")
+}
+
+function pushTail(sample, index, alias) {
+    let thisAlias = checkAliasCount(alias, false, false)
+
+    let start = offset + index * beatLength - 2*ovlVC
+    let ovl = start + ovlVC
+    let preu = start + 2*ovlVC
+    let fixed = start + 2*ovlVC + 10
+    let cutoff = 0 - (2*ovlVC + 20)
+    
+    let end = start - cutoff
+    let points = [fixed, preu, ovl]
+    points.push(start)
+    let extras = [cutoff.toString()]
+    let notes = new Notes()
+    if (appendTags) {
+        notes.tag = "Tail"
+    }
+    let entry = new Entry(sample, thisAlias, start, end, points, extras, notes)
+    push(entry, "VC")
+}
+
+function pushOther(sample) {
+    let alias = checkAliasCount(sample, true, false)
+
+    let start = offset
+    let ovl = start + 20
+    let preu = start + 10
+    let fixed = start + 30
+    let cutoff = 0 - (40)
+    
+    let end = start - cutoff
+    let points = [fixed, preu, ovl]
+    points.push(start)
+    let extras = [cutoff.toString()]
+    let notes = new Notes()
+    if (appendTags) {
+        notes.tag = "Others"
+    }
+    let entry = new Entry(sample, alias, start, end, points, extras, notes)
+    push(entry, "other")
 }
 
 function parseSample(sample) {
     if (prefix !== "" && !sample.startsWith(prefix)) {
-        pushCV(sample, 0, sample, true, "Others")
+        pushOther(sample)
         return
     }
 
-    let rest = (sample + appendSuffix).slice(prefix.length)
+    // Remove ".wav" extension before appending suffix
+    let rest = (sample.slice(0,-4) + appendSuffix).slice(prefix.length)
     let index = 0
     let lastVowel = ""
 
+    if(debug){
+        console.log('Sample: ' + rest)
+    }
+
     while (rest !== "") {
         let matched = texts.find(text => rest.startsWith(text))
         if (matched === undefined) {
             // handle suffix
             let suffix = suffixes.find(suffix => rest === suffix)
             if (suffix) {
                 let alias = (lastVowel + " " + suffix).trim()
-                pushCV(sample, index, alias, false, "Tail")
+                pushTail(sample, index, alias)
             } else if (index === 0) {
-                pushCV(sample, 0, sample, true, "Others")
+                pushOther(sample)
             }
             return
         }
 
+        let nextHasConsonant = false
+        let next = rest.slice(matched.length)
+        if (separator !== "" && next.startsWith(separator)){
+            next = next.slice(separator.length)
+        }
+        let nextMatched = texts.find(text => next.startsWith(text))
+        if (nextMatched !== undefined) {
+            let nextCons = map.get(nextMatched)[0]
```

Comment:
What's this `map` (I couldn't find where it's passed)? Do we need to check if `map.get(nextMatched)` has a value before getting its first element?

Reply:
`map` is created at Ln 78, and since `texts` is created from `map`, `map.get(nextMatched)` will definitely have a value when `nextMatched` is defined.

Reply:
I see. Could you give it a meaningful name?